$NetBSD: patch-ad,v 1.2 2000/01/03 23:36:20 fredb Exp $

This patch supplied by the author, Greg Banks <gnb@alphalink.com.au>,
in private mail.

--- glib_extra.c.orig	Sat Nov 13 19:23:07 1999
+++ glib_extra.c	Mon Jan  3 17:22:50 2000
@@ -153,6 +153,14 @@
 g_unix_source_destroy
 };
 
+/*
+ * TODO: there is some bizarre subtle problem with either my code
+ * or one of gtk or gdk which causes strange display bugs when
+ * this priority is made equal to 1. This needs to be properly
+ * tracked down and exterminated, but for the time being making
+ * the priority equal to G_PRIORITY_DEFAULT works -- Greg, 3Jan2000.
+ */
+#define G_UNIX_REAP_PRIORITY	G_PRIORITY_DEFAULT
 
 void
 g_unix_reap_init(void)
@@ -163,7 +171,7 @@
     {
     	first = FALSE;
 	g_source_add(
-		1,			/* priority */
+		G_UNIX_REAP_PRIORITY,	/* priority */
 		TRUE,			/* can_recurse */
 		&reaper_source_funcs,
 		(gpointer)0,		/* source_data */
